.github/workflows/ci.yaml

```diff
@@ -86,60 +51,156 @@ jobs:
       #      -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
       #      https://api.github.com/repos/${{ github.actor }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/timing
 
+
+  build_job:
+    timeout-minutes: 30
+    runs-on: ubuntu-20.04
+    needs: test_job
+
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+
+    permissions:      
+      contents: read  #permissions for publishing Docker images to Github Packages
+      packages: write #permissions for publishing Docker images to Github Packages
+    
+    defaults:
+      run:
+        working-directory: angular
+    
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Buld phase
+        run: npm run build
+
+      #---Build Docker image, Login & Publish to Docker Packages---
+      - name: Log in to the Container registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login $REGISTRY -u ${{ github.actor }} --password-stdin
+
+      - name: build Docker image on push event
+        if: github.event_name == 'push'
+        run: docker build -t $REGISTRY/$IMAGE_NAME:${GITHUB_SHA::7} -t $REGISTRY/$IMAGE_NAME:latest .
```

Comment:
- let's tag with first 7 symbols of SHA1 of the commit ID. 
- and also tag it with `latest`
then push both tags to remote registry

Reply:
done